deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+        self.answer_tuples = answer_tuples
+        self.background_tuples = background_tuples
+        self.object_handling = object_handling
+        self.use_context = use_context
+
+    def __str__(self):
+        to_return = 'MultipleTrueFalseTuplesWithBackgroundInstance: \n'
+        to_return += "  Answer Candidates: \n"
+        for answer_index in range(len(self.answer_tuples)):
+            string_answer_tuples = [str(a_tuple) for a_tuple in self.answer_tuples[answer_index]]
+            to_return += "\t [{0}] {1}\n".format(answer_index, ", \n".join(string_answer_tuples))
+        to_return += "\n  Background Tuples: \n " + ", \n".join(str(self.background_tuples))
+        return to_return
+
+    @overrides
+    def words(self) -> Dict[str, List[str]]:
+        words = defaultdict(list)
+        all_tuples = self.background_tuples + [curr_tuple for answer in self.answer_tuples
+                                               for curr_tuple in answer]
+        for text_tuple in all_tuples:
+            for phrase in text_tuple.to_text_list(self.object_handling,
+                                                  self.use_context):
+                phrase_words = self._words_from_text(phrase)
+                for namespace in phrase_words:
+                    words[namespace].extend(phrase_words[namespace])
+        return words
+
+    @overrides
+    def to_indexed_instance(self, data_indexer: DataIndexer):
+        answers_indexed = [self.index_tuples(curr_answer_tuples, data_indexer) for
+                           curr_answer_tuples in self.answer_tuples]
+        background_indexed = self.index_tuples(self.background_tuples, data_indexer)
+        return IndexedMCTuplesWithBackgroundInstance(answers_indexed,
+                                                     background_indexed,
+                                                     self.label,
+                                                     self.index)
+
+    # This is a helped method for to_indexed_instance which indexes a list of TextTuples
+    # Input:
+    #   tuples_in is a list of text tuples, each tuple is converted to a list of strings, then
+    #   each of the strings is converted to a list of indices.
+    # Return:
+    #   indexed: List[List[List[int]]]
+    def index_tuples(self, tuples_in: List[TextTuple], data_indexer: DataIndexer):
+        indexed = []
+        for text_tuple in tuples_in:
+            text_list = text_tuple.to_text_list(self.object_handling, self.use_context)
+            # indices: List[List[int]]
+            indices = [self._index_text(phrase, data_indexer) for phrase in text_list]
+            indexed.append(indices)
+        return indexed
+
+    @classmethod
+    def read_from_line(cls, line: str, default_label: int=None):
+        """
+        Reads a TupleInstances from a line.  The format has one option (currently):
+            [question index][tab][num answer coptions][tab][all question+answer tuples][tab]
```

Comment:
With the number of answer options in the third column.  This is related to my earlier comment about encoding the same information twice.